src/charm.py

```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/env python3
+
+"""Subordinate charm for TimescaleDB."""
+import os
+from subprocess import PIPE, Popen, check_call, check_output
+
+from ops.charm import CharmBase
+from ops.framework import StoredState
+from ops.main import main
+from ops.model import ActiveStatus, BlockedStatus, WaitingStatus
+
+
+class TimescaleDB(CharmBase):
+    """Subordinate charm for TimescaleDB."""
+
+    state = StoredState
```

Comment:
It's usually advised to avoid StoredState if you can help it. This [doc](https://juju.is/docs/sdk/stored-state-uses-limitations) provides good context and alternatives. Using peer relations is usually the way to go for something like this and I believe the Contracts Core charm has a nice construct to easily access the peer relation.

Reply:
Thanks for the advice, I'll look into it. This was the way the initial charm was built and I did not touch that part yet...

Reply:
Don't you need to instantiate it in any case?